app/scripts/views/complete_account_unlock.js

```diff
@@ -0,0 +1,90 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+define([
+  'views/form',
+  'views/base',
+  'stache!templates/complete_account_unlock',
+  'lib/auth-errors',
+  'lib/validate'
+],
+function (FormView, BaseView, CompleteAccountUnlockTemplate, AuthErrors,
+    Validate) {
+
+  function doesLinkValidate(uid, code, isLinkDamaged) {
+    return Validate.isUidValid(uid) &&
+           Validate.isCodeValid(code) &&
+           ! isLinkDamaged;
+  }
+
+  var CompleteAccountUnlockView = FormView.extend({
+    template: CompleteAccountUnlockTemplate,
+    className: 'complete_account_unlock',
+
+    beforeRender: function () {
+      try {
+        this.importSearchParam('uid');
+        this.importSearchParam('code');
+      } catch(e) {
+        this.logScreenEvent('link_damaged');
+        // This is an invalid link. Abort and show an error message
+        // before doing any more checks.
+        return true;
+      }
+
+      // Remove any spaces that are probably due to a MUA adding
+      // line breaks in the middle of the link.
+      this.uid = this.uid.replace(/ /g, '');
+      this.code = this.code.replace(/ /g, '');
+
+      if (! doesLinkValidate(this.uid, this.code, this._isLinkDamaged)) {
+        // One or more parameters fails validation. Abort and show an
+        // error message before doing any more checks.
+        this.logScreenEvent('link_damaged');
+        return true;
+      }
+
+      var self = this;
+      return this.fxaClient.completeAccountUnlock(this.uid, this.code)
+        .then(function () {
+          self.logScreenEvent('verification.success');
+          self.navigate('account_unlock_complete');
+          return false;
+        })
+        .fail(function (err) {
+          if (AuthErrors.is(err, 'UNKNOWN_ACCOUNT')) {
+            self._isLinkExpired = true;
+            self.logScreenEvent('link_expired');
+          } else if (AuthErrors.is(err, 'INVALID_VERIFICATION_CODE') ||
+              AuthErrors.is(err, 'INVALID_PARAMETER')) {
+            // These errors show a link damaged screen
+            self._isLinkDamaged = true;
+            self.logScreenEvent('link_damaged');
+          } else {
+            // all other errors show the standard error box.
+            self._error = self.translateError(err);
+          }
+          return true;
+        });
+    },
+
+    context: function () {
+      var isLinkValid = doesLinkValidate(
+              this.uid, this.code, this._isLinkDamaged);
+      var isLinkExpired = this._isLinkExpired;
+
+      return {
+        // If the link is invalid, print a special error message.
+        isLinkDamaged: ! isLinkValid,
```

Comment:
Slight naming inconsistency here with `this._isLinkDamaged` and `isLinkDamanged` meaning different things.  Could use `isLinkInvalid` here for consistency?